pkg/lib/bundlev2/synthesis.go

```diff
@@ -0,0 +1,375 @@
+package bundlev2
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	cv1 "k8s.io/api/core/v1"
+	rv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+const (
+	DefaultPermission = 0644
+	// MediaType
+	RegistryV2Type = "registry+v2"
+	PlainType      = "plain"
+	HelmType       = "helm"
```

Comment:
I think we might already define this type somewhere else in the codebase - if so can we use that instead?

Reply:
yes, I deliberately rewrote it again. These exist in bundle+v1. I assume, redefining them in bundle+v2 is necessary.

Reply:
Maybe we should have these types in a higher level package somewhere and then import into both v1/v2? @kevinrizza

Reply:
Oh, i just thought they might be different. at least someday. I am not even using them, i will delete them for now. If we think they will always be the same for v1 and v2, then we should move this out of bundle v1.